### PR TITLE
Update contributing.rst

### PR DESCRIPTION
**Pull request by sunqm**
Update url of extension-template

### CODE REVIEW

File: source/contributing.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ How to make an extension
 1. Download the template (https://github.com/pyscf/extension-template), run the
    script in the extension template::
 
-    $ curl https://github.com/pyscf/extension-template/pyscf-extension-template-0.1.0.tar.gz | tar xvzf -
+    $ curl https://github.com/pyscf/extension-template/archive/refs/tags/0.1.0.tar.gz | tar xvzf -
```
Comment (susilehtola) [inline source/contributing.rst]: Neither of these works for me
```
$ curl https://github.com/pyscf/extension-template/archive/refs/tags/0.1.0.tar.gz | tar xvzf -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   141  100   141    0     0    819      0 --:--:-- --:--:-- --:--:--   819

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
$ curl https://github.com/pyscf/extension-template/pyscf-extension-template-0.1.0.tar.gz | tar xvzf -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     9  100     9    0     0    102      0 --:--:-- --:--:-- --:--:--   103

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```

Comment (bogdanoff) [reply source/contributing.rst]: `curl -L` will follow the redirection done by github

Comment (susilehtola) [reply source/contributing.rst]: @bogdanoff great, that fixes it

Comment (susilehtola) [inline source/contributing.rst]:
```suggestion
    $ curl -L https://github.com/pyscf/extension-template/archive/refs/tags/0.1.0.tar.gz | tar xvzf -
```

**Review by susilehtola — APPROVED** [review]

**Pull request merged**